api/preprint_providers/views.py

```diff
@@ -263,6 +263,25 @@ def get_queryset(self):
         return provider.all_subjects
 
 
+class PreprintProviderHighlightedSubjectList(JSONAPIBaseView, generics.ListAPIView):
+    permission_classes = (
+        drf_permissions.IsAuthenticatedOrReadOnly,
+        base_permissions.TokenHasScope,
+    )
+
+    view_category = 'preprint_providers'
+    view_name = 'highlighted-taxonomy-list'
+
+    required_read_scopes = [CoreScopes.ALWAYS_PUBLIC]
+    required_write_scopes = [CoreScopes.NULL]
+
+    serializer_class = TaxonomySerializer
+
+    def get_queryset(self):
+        provider = get_object_or_error(PreprintProvider, self.kwargs['provider_id'], display_name='PreprintProvider')
+        return provider.highlighted_subjects
```

Comment:
I think if filtering happens after `get_queryset` and filtering is written to expect a queryset then this could esplode because `provider.highlighted_subjects` might return a list and `list.filter(thing=stuff)` is going to throw an exception. Most places in the api I forced returning a queryset by doing a new query for the ids of the things after a `.values_list('id', flat=True)`. It's kind of bad but it's less bad than an exception.